AmplifyPlugins/DataStore/AWSDataStoreCategoryPluginIntegrationTests/DataStoreConsecutiveUpdatesTests.swift

```diff
@@ -0,0 +1,331 @@
+                XCTAssertNil(post)
+                queryExpectation.fulfill()
+            case .failure(let error):
+                XCTFail("Error: \(error)")
+            }
+        }
+        wait(for: [queryExpectation], timeout: networkTimeout)
+
+    }
+
+    /// - Given: API has been setup with `Post` model registered
+    /// - When: A Post is saved with sync complete and updated
+    /// - Then: The post should be updated with new fields
+    func testSaveAndUpdate() throws {
+        try startAmplifyAndWaitForSync()
+
+        // create a post
+        let myPost = Post(id: UUID().uuidString,
+                          title: "MyPost",
+                          content: "This is my post.",
+                          createdAt: Temporal.DateTime.now(),
+                          rating: 3,
+                          status: .published)
+
+        let saveSyncExpectation = expectation(description: "Post is saved and synced")
+        let updateSyncExpectation = expectation(description: "Post is updated and synced")
+
+        let hubListener = Amplify.Hub.listen(
+            to: .dataStore,
+            eventName: HubPayload.EventName.DataStore.syncReceived) { payload in
+            guard let mutationEvent = payload.data as? MutationEvent
+            else {
+                XCTFail("Can't cast payload as mutation event")
+                return
+            }
+
+            guard let post = try? mutationEvent.decodeModel() as? Post,
+                  post.id == myPost.id else {
+                return
+            }
+
+            if mutationEvent.mutationType == GraphQLMutationType.create.rawValue {
+                XCTAssertEqual(post.title, "MyPost")
+                XCTAssertEqual(post.content, "This is my post.")
+                XCTAssertEqual(post.rating, 3)
+                saveSyncExpectation.fulfill()
+                return
+            }
+
+            if mutationEvent.mutationType == GraphQLMutationType.update.rawValue {
+                XCTAssertEqual(post.title, "MyUpdatedPost")
+                XCTAssertEqual(post.content, "This is my updated post.")
+                XCTAssertEqual(post.rating, 5)
+                updateSyncExpectation.fulfill()
+                return
+            }
+        }
+
+        guard try HubListenerTestUtilities.waitForListener(with: hubListener, timeout: 5.0) else {
+            XCTFail("Listener not registered for hub")
+            return
+        }
+
+        // save the post and wait for it to sync
+        let immediateSaveExpectation = expectation(description: "Post is saved")
+        Amplify.DataStore.save(myPost) { result in
+            switch result {
+            case .success:
+                immediateSaveExpectation.fulfill()
+            case .failure(let error):
+                XCTFail("Error: \(error)")
+            }
+        }
+        wait(for: [immediateSaveExpectation], timeout: networkTimeout)
+        wait(for: [saveSyncExpectation], timeout: networkTimeout)
```

Comment:
no need to wait for it to sync if you are trying to test "Save and immediately delete it". in this scenario, the eventual consistent state should be item is deleted